src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -181,21 +186,20 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
             }
         }
 
-        private bool GetCanIncremental(string versionName)
+        private bool GetCanIncremental(string configHash, string templateHash, string versionName)
         {
             if (LastBuildInfo == null)
             {
                 return false;
             }
-            if (CurrentBuildInfo.DocfxVersion != LastBuildInfo.DocfxVersion ||
-                CurrentBuildInfo.PluginHash != LastBuildInfo.PluginHash ||
-                CurrentBuildInfo.TemplateHash != LastBuildInfo.TemplateHash ||
-                CurrentBuildInfo.ConfigHash != LastBuildInfo.ConfigHash)
+            var version = LastBuildInfo.Versions.SingleOrDefault(v => v.VersionName == versionName);
+            if (version == null || configHash != version.ConfigHash || templateHash != version.TemplateHash)
```

Comment:
right now not all properties of BuildVersionInfo could be used for check CanIncremental

---
In reply to: [75423251](https://github.com/dotnet/docfx/pull/621#discussion_r75423251) [](ancestors = 75423251)